### PR TITLE
libc: fix CONFIG_STDOUT_CONSOLE semantics

### PR DESCRIPTION
**Pull request by tyrantlizard77**
The intent of this Kconfig is to allow libc stdout
functions like printf() to send their output to the
active console driver instead of discarding it.

This somehow evolved into preferring to use
printf() instead of printk() for all test case output
if enabled. Libc printf() implementation for both
minimal libc and newlib use considerably more stack
space than printk(), with nothing gained by using
them.

Remove all instances where we are conditionally
sending test case output based on this config, enable
it by default, and adjust a few tests that disabled
this because they were blowing stack.

Fixes: #13701

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Comment by andrewboie** [comment]
ugh, the "unit_testing" target unfortunately relies on this behavior.
@nashif unit_testing is only built for 3 tests, what is this for? Can we get rid of it? It seems to be a giant hack that ought to be superseded by native_posix

File: lib/libc/Kconfig [hunk 1]
```diff
@@ -47,7 +47,7 @@ config NEWLIB_LIBC_FLOAT_SCANF
 config STDOUT_CONSOLE
 	bool "Send stdout to console"
 	depends on CONSOLE_HAS_DRIVER
-	default NEWLIB_LIBC
+	default y
```
Comment (pfalcon) [inline lib/libc/Kconfig]: Why this change specifically?

**Comment by pfalcon** [comment]
As a note, the commit message here talks about "test case output", but patch also updates *samples*. That said, the description of CONFIG_STDOUT_CONSOLE is quite clear, and the patch looks like a good cleanup. So, informal +1 from me, until build breakage issues are understood/solved.

File: lib/libc/Kconfig (hunk 1 above)
Comment (andrewboie) [reply lib/libc/Kconfig]: having libc stdout functions do what is expected seems a reasonable default

**Review by dcpleung — APPROVED** [review]

**Comment by nashif** [comment]
> @nashif unit_testing is only built for 3 tests, what is this for? Can we get rid of it? It seems to be a giant hack that ought to be superseded by native_posix

is on TODO,, not sure if we have external users, need to check. @carlescufi / @SebastianBoe  do we have anyone using this?

**Comment by andrewboie** [comment]
> is on TODO,, not sure if we have external users, need to check. @carlescufi / @SebastianBoe do we have anyone using this?

I fixed the issue wrt this patch, needed to add mock functions for printk/vprintk and now it works.

**Review by pfalcon — APPROVED** [review]

**Comment by SebastianBoe** [comment]
> is on TODO,, not sure if we have external users, need to check. @carlescufi / @SebastianBoe do we have anyone using this?

Checking internally ...

> Can we get rid of it? It seems to be a giant hack that ought to be superseded by native_posix

IIRC the unittest.cmake would need to be ported/re-written to use native_posix somehow, native_posix can not replace unittest.cmake as-is (it doesn't have a unit test framework).

**Comment by SebastianBoe** [comment]
AFAICT Nordic doesn't use unittest.cmake in a signficant way, only ztest.h.

So it should be safe on our side to drop support for it.

**Review by galak — APPROVED** [review]

**Comment by SebastianBoe** [comment]
... is unittest.cmake broken now?
